src/services/account/useInvoices.ts

```diff
@@ -0,0 +1,150 @@
+import { useQuery } from '@tanstack/react-query'
+import { z } from 'zod'
+
+import Api from 'shared/api'
+import { NetworkErrorObject } from 'shared/api/helpers'
+
+const DefaultPaymentMethodSchema = z
+  .object({
+    billingDetails: z.object({
+      address: z.object({
+        city: z.string().nullable(),
+        country: z.string().nullable(),
+        line1: z.string().nullable(),
+        line2: z.string().nullable(),
+        postalCode: z.string().nullable(),
+        state: z.string().nullable(),
+      }),
+      email: z.string().nullable(),
+      name: z.string().nullable(),
+      phone: z.string().nullable(),
+    }),
+    card: z.object({
+      brand: z.string().nullable(),
+      expMonth: z.number().nullable(),
+      expYear: z.number().nullable(),
+      last4: z.string().nullable(),
+    }),
+  })
+  .nullable()
+
+const InvoiceSchema = z.object({
+  amountDue: z.number(),
+  amountPaid: z.number(),
+  created: z.number(),
+  currency: z.string().nullable(),
+  customerAddress: z.string().nullable(),
+  customerEmail: z.string().nullable(),
+  customerName: z.string().nullable(),
+  defaultPaymentMethod: DefaultPaymentMethodSchema,
+  dueDate: z.number().nullish(),
```

Comment:
qq what's the diff between nullish and nullable?

Reply:
`nullish === null | undefined`, which should never occur in GQL, we should stick to `nullable` only.